src/modules/flow/converter/string-format.c

```diff
@@ -594,14 +591,16 @@ insert_numbers(struct sol_buffer *out,
     SOL_INT_CHECK(r, < 0, r);
 
     if (n_zeros) {
-        char *str = alloca(n_zeros + 1);
+        char *str;
 
-        SOL_NULL_CHECK(str, -ENOMEM);
-        r = snprintf(str, n_zeros + 1, "%0*d", (int)(long)n_zeros, 0);
-        SOL_INT_CHECK(r, < 0, r);
+        r = asprintf(&str, "%0*d", (int)(long)n_zeros, 0);
+        SOL_INT_CHECK(r, < 0, -ENOMEM);
+
+        slice = SOL_STR_SLICE_STR(str, r);
+        r = sol_buffer_insert_slice(out, pos, slice);
```

Comment:
no need to `asprintf()` and then `sol_buffer_insert_slice()`, we should have `sol_buffer_insert_printf()`. @glima, please review this whole file as you did the port, need to take care to use `sol_buffer` infra as much as possible.